### PR TITLE
Improve syntax lookup history behaviour

### LINKED ISSUE #383: Back button behaviour in Syntax lookup
When I open something in the excellent Syntax lookup I would expect that the back button brings me back to the overview.
But right now you need to click the `X` to close the popup, pressing the back button will bring you to the page you visited before the syntax page.

Example: https://cln.sh/AcT8y6

Not a major issue, maybe it's just me who thinks this is weird?

### PR DESCRIPTION
**Pull request by Minnozz**
(I based this PR on #387 because my development setup was not usable otherwise)

- Use `Next.Router.push` instead of `replace` to allow the back button to work
- Correctly link URL state to component state using [a reducer with side-effects](https://github.com/bloodyowl/rescript-react-update)

Fixes #383

### CODE REVIEW

**Comment by Minnozz** [comment]
I was hesitant to add a dependency to this project, but I think this is the cleanest way to write this logic.

**Comment by ryyppy** [comment]
What makes the logic cleaner with ReactUpdate? It might look cleaner on first glance, but for me it's currently not entirely clear what benefits I get over the approach we had before?

File: src/SyntaxLookup.res [hunk 1]
```diff
@@ -153,75 +153,64 @@ type state =
...
-      } else {
-        ()
-      }
-    | (ShowDetails(item), None) =>
-      router->Next.Router.replace("syntax-lookup#" ++ GithubSlugger.slug(item.id))
-    | (_, Some(_)) => router->Next.Router.replace("syntax-lookup")
-    | _ => ()
-    }
-    None
-  }, [state])
```
Comment (Minnozz) [inline src/SyntaxLookup.res:198]: @ryyppy For me, the improvement is being able to connect a specific action (like clearing the search box) to this effect (updating the URL). With ReactUpdate, you can queue the effect at the same location where you handle the action, whereas in the old version you have to compare the current URL and new state to reverse engineer what the most recent change was.

**Comment by Minnozz** [comment]
See inline comment: https://github.com/rescript-association/rescript-lang.org/pull/385#discussion_r650729738

Of couse, what is clearer is subjective, and if you want I can try to achieve the same behaviour without this library.

**Comment by ryyppy** [comment]
@Minnozz Now that I had a closer look at rescript-react-update, I think your solution is valid, but for now I wanted to keep the dependency footprint low. #445 fixes the router problem (including a few other UX issues), so this should be fixed now.

Sorry for my late response times; I try to have a closer eye on open PRs now, in case you see potential improvements in the code.

**Pull request closed without merging**